azurerm_storage_management_policy/main.tf

```diff
@@ -0,0 +1,38 @@
+provider "azurerm" {
+  version = "=2.4.0"
+  features {}
+}
+
+terraform {
+  # The configuration for this backend will be filled in by Terragrunt
+  backend "azurerm" {}
+}
+
+
+resource "azurerm_storage_management_policy" "storage_management_policy" {
+  storage_account_id = var.storage_account_id
+
+  dynamic "rule" {
+    for_each = var.rules
```

Comment:
Have you tested what happens changing the rules order?

Reply:
it simply applies changes. It does not recreate the rules. 
Below an example of the plan:

```
 ~ rule {
            enabled = true
          ~ name    = "deleteafter3yrs" -> "deleteafter2yrs"

          ~ actions {
              ~ base_blob {
                  ~ delete_after_days_since_modification_greater_than          = 1095 -> 730
                    tier_to_archive_after_days_since_modification_greater_than = 0
                    tier_to_cool_after_days_since_modification_greater_than    = 0
                }

                snapshot {
                    delete_after_days_since_creation_greater_than = 0
                }
            }

          ~ filters {
                blob_types   = [
                    "blockBlob",
                ]
              ~ prefix_match = [
                  - "userdata/*",
                  + "usermessages/*",
                ]
            }
        }
      ~ rule {
            enabled = true
          ~ name    = "deleteafter2yrs" -> "deleteafter3yrs"

          ~ actions {
              ~ base_blob {
                  ~ delete_after_days_since_modification_greater_than          = 730 -> 1095
                    tier_to_archive_after_days_since_modification_greater_than = 0
                    tier_to_cool_after_days_since_modification_greater_than    = 0
                }

                snapshot {
                    delete_after_days_since_creation_greater_than = 0
                }
            }

          ~ filters {
                blob_types   = [
                    "blockBlob",
                ]
              ~ prefix_match = [
                  + "userdata/*",
                  - "usermessages/*",
                ]
            }
        }
    }
```

Reply:
As you see changing the order of the rule recreate also existing rule.

Reply:
It's a change in place. How can I do it instead? Changing these rules it's not causing any inefficiency I guess.